### PR TITLE
DEPR: rename Timestamp.offset to .freq

### LINKED ISSUE #12160: DEPR: rename Timestamp.offset internally .freq / deprecate passing offset in favor of freq
xref  #6815/#6813 .freq in Timestamp rather than .offset (internally used) (0.14.0 ???)

This is a bit non-trivial so pulling out to a separate issue from #6581

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] closes #12160
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/tests/test_packers.py [hunk 1]
```diff
@@ -58,6 +58,19 @@ def check_arbitrary(a, b):
         assert_series_equal(a, b)
     elif isinstance(a, Index):
         assert_index_equal(a, b)
+    elif isinstance(a, Categorical):
+        # Temp,
+        # Categorical.categories is changed from str to bytes in PY3
+        # maybe the same as GH 13591
+        if PY3 and b.categories.inferred_type == 'string':
```
Comment (sinhrks) [inline pandas/io/tests/test_packers.py]: seems to be the same issue as #13591

Comment (jreback) [inline pandas/io/tests/test_packers.py]: cc @kawochen

File: pandas/src/period.pyx [hunk 2]
```diff
@@ -484,8 +484,11 @@ def extract_freq(ndarray[object] values):
 
     for i in range(n):
         p = values[i]
+
         try:
-            return p.freq
+            # now Timestamp / NaT has freq attr
```
Comment (jreback) [inline pandas/src/period.pyx]: I though we have an ``is_period`` in ``inference.pyx``? (it does this)

File: pandas/tseries/index.py [hunk 3]
```diff
@@ -1199,8 +1199,9 @@ def __iter__(self):
         for i in range(chunks):
             start_i = i * chunksize
             end_i = min((i + 1) * chunksize, l)
-            converted = tslib.ints_to_pydatetime(
-                data[start_i:end_i], tz=self.tz, offset=self.offset, box=True)
+            converted = tslib.ints_to_pydatetime(data[start_i:end_i],
+                                                 tz=self.tz, freq=self.offset,
```
Comment (jreback) [inline pandas/tseries/index.py]: shouldn't this be ``self.freq``?

File: pandas/tslib.pyx [hunk 4]
```diff
@@ -338,25 +342,34 @@ class Timestamp(_Timestamp):
         cdef _TSObject ts
         cdef _Timestamp ts_base
 
+        if offset is not None:
+            # deprecate offset arg
```
Comment (jreback) [inline pandas/tslib.pyx]: put the issue reference

File: pandas/tslib.pyx [hunk 5]
```diff
@@ -911,16 +926,6 @@ cdef inline bint _is_multiple(int64_t us, int64_t mult):
     return us % mult == 0
 
 
-def apply_offset(ndarray[object] values, object offset):
-    cdef:
```
Comment (jreback) [inline pandas/tslib.pyx]: not used?

Comment (sinhrks) [inline pandas/tslib.pyx]: Not used, and the definition is meaningless.

File: pandas/tseries/index.py (hunk 3 above)
Comment (sinhrks) [inline pandas/tseries/index.py]: Fixed. Though ``DatetimeIndex.freq`` is a mapping to ``offset``.

File: pandas/io/tests/test_packers.py [hunk 6]
```diff
@@ -58,6 +58,19 @@ def check_arbitrary(a, b):
...
+            pass
+        else:
+            tm.assert_categorical_equal(a, b)
+    elif a is NaT:
+        assert b is NaT
```
Comment (jreback) [inline pandas/io/tests/test_packers.py]: we should prob change the comparators to be like the rest of pandas (e.g. assertTrue).... but can do later

File: pandas/src/period.pyx (hunk 2 above)
Comment (jreback) [inline pandas/src/period.pyx]: right I see what we did with this now. ok

**Comment by jreback** [comment]
thanks @sinhrks 

I had to do a couple of more renames to avoid the warnings...but no biggie.

not that ``Timestamp.from_ordinal`` also needed changing, but I don't think its tested much.

**Comment by sinhrks** [comment]
Thanks, let me do a follow-up for testing ``fromordinal``.

**Pull request closed without merging**